store/sqlstore/post_store.go

```diff
@@ -474,7 +474,23 @@ func (s *SqlPostStore) getPostWithCollapsedThreads(id, userID string, extended b
 
 		return nil, errors.Wrapf(err, "failed to get Post with id=%s", id)
 	}
-	return s.prepareThreadedResponse([]*postWithExtra{&post}, extended, false)
+
+	var posts []*model.Post
+	_, err = s.GetReplica().Select(&posts, "SELECT * FROM Posts WHERE Posts.RootId = :RootId AND DeleteAt = 0", map[string]interface{}{"RootId": id})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find Posts")
+	}
```

Comment:
Would it be helpful to check if err is `sql.ErrNoRows`? Also how about adding `id` to the error message for more context.

Reply:
@ashishbhate  there should be no case for 'no rows' , since this is a fetch for posts in a thread. If there are no rows - it's an error.

Reply:
i will add id to the error msg